Userland/Libraries/LibWeb/Bindings/LegacyPlatformObject.h

```diff
@@ -25,16 +27,70 @@ class LegacyPlatformObject : public PlatformObject {
     virtual JS::ThrowCompletionOr<JS::MarkedVector<JS::Value>> internal_own_property_keys() const override;
 
     JS::ThrowCompletionOr<bool> is_named_property_exposed_on_object(JS::PropertyKey const&) const;
-    JS::ThrowCompletionOr<Optional<JS::PropertyDescriptor>> legacy_platform_object_get_own_property_for_get_own_property_slot(JS::PropertyKey const&) const;
-    JS::ThrowCompletionOr<Optional<JS::PropertyDescriptor>> legacy_platform_object_get_own_property_for_set_slot(JS::PropertyKey const&) const;
+
+    enum class IgnoreNamedProps {
+        No,
+        Yes,
+    };
+    JS::ThrowCompletionOr<Optional<JS::PropertyDescriptor>> legacy_platform_object_get_own_property(JS::PropertyKey const&, IgnoreNamedProps ignore_named_props) const;
 
     virtual JS::Value item_value(size_t index) const;
     virtual JS::Value named_item_value(DeprecatedFlyString const& name) const;
     virtual Vector<DeprecatedString> supported_property_names() const;
     virtual bool is_supported_property_index(u32) const;
 
+    // NOTE: These will crash if you make has_named_property_setter return true but do not override these methods.
+    // NOTE: This is only used if named_property_setter_has_identifier returns false, otherwise set_value_of_named_property is used instead.
+    virtual WebIDL::ExceptionOr<void> set_value_of_new_named_property(DeprecatedString const&, JS::Value) { TODO(); }
+    virtual WebIDL::ExceptionOr<void> set_value_of_existing_named_property(DeprecatedString const&, JS::Value) { TODO(); }
+
+    // NOTE: These will crash if you make has_named_property_setter return true but do not override these methods.
+    // NOTE: This is only used if you make named_property_setter_has_identifier return true, otherwise set_value_of_{new,existing}_named_property is used instead.
+    virtual WebIDL::ExceptionOr<void> set_value_of_named_property(DeprecatedString const&, JS::Value) { TODO(); }
+
+    // NOTE: These will crash if you make has_indexed_property_setter return true but do not override these methods.
+    // NOTE: This is only used if indexed_property_setter_has_identifier returns false, otherwise set_value_of_indexed_property is used instead.
+    virtual WebIDL::ExceptionOr<void> set_value_of_new_indexed_property(u32, JS::Value) { TODO(); }
+    virtual WebIDL::ExceptionOr<void> set_value_of_existing_indexed_property(u32, JS::Value) { TODO(); }
+
+    // NOTE: These will crash if you make has_named_property_setter return true but do not override these methods.
+    // NOTE: This is only used if indexed_property_setter_has_identifier returns true, otherwise set_value_of_{new,existing}_indexed_property is used instead.
+    virtual WebIDL::ExceptionOr<void> set_value_of_indexed_property(u32, JS::Value) { TODO(); }
```

Comment:
`TODO` is a placeholder, but there won't be anything added here -> `VERIFY_NOT_REACHED`